src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -103,9 +128,44 @@ public void launch() {
             kernel.getContext().get(c).startup();
         }
 
+        final Queue<DeviceIdentityInterface> provisioningPlugins = findProvisioningPlugins();
         // Must be called before everything else so that these are available to be
         // referenced by main/dependencies of main
         final Queue<String> autostart = findBuiltInServicesAndPlugins(); //NOPMD
+        // run the provisioning if device is not provisioned
+        if (!kernel.getContext().get(DeviceConfiguration.class).isDeviceConfiguredToTalkToCloud()
```

Comment:
You should be checking the provisioning policy here right? If customers don't want to provision then we should skip this.

Reply:
At this point in nucleus, I am not sure if we will even have a provisioning policy set because if device is not connected then there is no deployment from cloud which is where the policy is specified. 

In cases where provisioning policy is available, I pass the policy to the plugin so that it can make a decision of what it wants to do based on the policy, as opposed to nucleus making that decision. If they do not want to provision, they can return empty configuration.